src/caselawclient/search_result.py

```diff
@@ -0,0 +1,226 @@
+            node.xpath(
+                "search:extracted/akn:FRBRdate[@name='transform']/@date",
+                namespaces=namespaces,
+            )
+            or [""]
+        )[0]
+        content_hash = (
+            node.xpath("search:extracted/uk:hash/text()", namespaces=namespaces) or [""]
+        )[0]
+
+        try:
+            date = dateparser.parse(date_string)
+        except ParserError as e:
+            logging.warning(
+                f'Unable to parse document date "{date_string}". Full error: {e}'
+            )
+            date = None
+
+        try:
+            court = courts.get_by_code(court_code)
+        except CourtNotFoundException:
+            court = None
+
+        return SearchResult(
+            uri=uri,
+            neutral_citation=neutral_citation,
+            name=metadata_name,
+            matches=matches_html,
+            court=court,
+            date=date,
+            content_hash=content_hash,
+            transformation_date=transformation_date,
+            meta=meta,
+        )
+
+
+class EditorStatus(Enum):
+    """
+    Enum representing the editor status.
+    """
+
+    NEW = "new"
+    IN_PROGRESS = "in progress"
+    HOLD = "hold"
+
+
+@dataclass
+class SearchResultMeta:
+    """
+    Represents the metadata of a search result.
+
+    Attributes:
+        author (str): The author of the search result.
+        author_email (str): The email of the author.
+        consignment_reference (str): The consignment reference.
+        submission_datetime (Optional[datetime]): The submission datetime.
+        assigned_to (str): The assigned editor.
+        editor_hold (str): The editor hold status.
+        editor_priority (str): The editor priority.
+        last_modified (str): The last modified date.
+
+    Properties:
+        editor_status (EditorStatus): The editor status based on the metadata.
+    """
+
+    author: str = ""
+    author_email: str = ""
+    consignment_reference: str = ""
+    submission_datetime: Optional[datetime] = None
+    assigned_to: str = ""
+    editor_hold: str = "false"
+    editor_priority: str = ""
+    last_modified: str = ""
+
+    @property
+    def editor_status(
+        self,
+    ) -> EditorStatus:
+        """
+        Get the editor status based on the metadata.
+
+        Returns:
+            EditorStatus: The editor status.
+        """
+        if self.editor_hold == "true":
+            return EditorStatus.HOLD
+        if self.assigned_to:
+            return EditorStatus.IN_PROGRESS
+        return EditorStatus.NEW
+
+    @staticmethod
+    def create_from_uri(uri: str) -> "SearchResultMeta":
+        """
+        Create a SearchResultMeta instance from a URI.
+
+        Args:
+            uri (str): The URI of the search result.
+
+        Returns:
+            SearchResultMeta: The created SearchResultMeta instance.
+        """
+        response_text = api_client.get_properties_for_search_results([uri])
```

Comment:
The SearchResultMeta class is currently passed to the SearchResult class through dependency injection.

This is because it needs to make calls to marklogic with `api_client`.

In my mind this is just a stop gap, but ideally I feel like we could return the "metadata" of a document as part of the `api_client.advanced_search`  (Marklogic `search_v2.xqy`) call. At the moment, to parse the results, we are having to make n extra calls for every advanced_search call where n is the number of results in the search which is inefficient. On top of that, we wouldnt need this extra code!

What do you guys think?

@dragon-dxw @jacksonj04